src/main/java/com/shapesecurity/salvation/directiveValues/MediaType.java

```diff
@@ -25,6 +25,10 @@ public boolean matchesMediaType(@Nonnull MediaType mediaType) {
         return this.type.equalsIgnoreCase(mediaType.type) && this.subtype.equalsIgnoreCase(mediaType.subtype);
     }
 
+    public boolean matchesTypeOrSubType(@Nonnull String str) {
```

Comment:
I don't like this function. It's not generally useful. Just use `x.type.equals("*") || x.subtype.equals("*")`.